request-manager/graph/grapher.go

```diff
@@ -177,25 +189,32 @@ func getMissingSets(actual map[string]bool, declared []*spec.NodeSet) []string {
 	return missing
 }
 
-// Builds a sequence graph and performs checks on that sequence.
+// buildSeqGraph builds a sequence graph and performs checks on that sequence.
 // It does not analyze any subsequences. As such, it can't e.g. check validity
 // of `sets` declarations in sequence and conditional nodes.
-// Returns the graph, the set of job args its component nodes set, and an error if any should occur.
+// Returns the graph, the set of job args its component nodes set, and an error
+// if any should occur.
 func buildSeqGraph(seqSpec *spec.Sequence, idgen id.Generator) (seqGraph *Graph, sets map[string]bool, err error) {
 	// The graph we'll be filling in
-	seqGraph, err = newGraph(seqSpec.Name, idgen)
+	seqGraph, err = newSeqGraph(seqSpec.Name, idgen)
 	if err != nil {
 		return nil, nil, err
 	}
-	// newGraph doesn't manager `Order` for us; we'll have to maintain it ourselves in this function
+
+	// We have to maintain `Order` ourselves
 	seqGraph.Order = append(seqGraph.Order, seqGraph.Source)
```

Comment:
I think the next iteration of `type Graph struct` should do this internally. Order is a property of the Graph, so the Graph should maintain it, leaving the caller to simply use the Graph without worrying about internal, low-level graph details.